war/src/main/java/io/lumeer/engine/controller/CollectionMetadataFacade.java

```diff
@@ -176,236 +184,118 @@ public String createInternalName(String originalCollectionName) throws UserColle
-         attributes.add(name);
-      }
-
-      return attributes;
+   public Set<String> getAttributesNames(String collectionName) {
+      return getAttributesInfo(collectionName).keySet();
    }
 
    /**
-    * Gets complete info about collection attributes
+    * Gets complete info about collection attributes.
     *
     * @param collectionName
     *       internal collection name
-    * @return list of DataDocuments, each with info about one attribute
-    * @throws CollectionNotFoundException
-    *       when metadata collection is not found
+    * @return map, keys are attributes' names, values are objects with attributes info
     */
-   public List<DataDocument> getCollectionAttributesInfo(String collectionName) throws CollectionNotFoundException {
-      String metadataCollectionName = collectionMetadataCollectionName(collectionName);
-      checkIfMetadataCollectionExists(metadataCollectionName);
-
-      return dataStorage.run(queryDocumentFromCollectionMetadata(collectionName, LumeerConst.Collection.COLLECTION_ATTRIBUTES_META_TYPE_VALUE));
+   public Map<String, Attribute> getAttributesInfo(String collectionName) {
+      return getCollectionMetadata(collectionName).getAttributes();
    }
 
    /**
     * Renames existing attribute in collection metadata.
-    * This method should be called only when also renaming attribute in documents,
-    * and access rights should be checked there so they are not checked twice.
+    * This method should be called only when also renaming attribute in all collection documents.
     *
     * @param collectionName
     *       internal collection name
     * @param oldName
     *       old attribute name
     * @param newName
     *       new attribute name
-    * @return true if rename is successful, false if some metadata do not exist and the rename could not be performed
     * @throws AttributeAlreadyExistsException
     *       when attribute with new name already exists
     */
-   public boolean renameCollectionAttribute(String collectionName, String oldName, String newName) throws AttributeAlreadyExistsException {
-      String metadataCollectionName = collectionMetadataCollectionName(collectionName);
-
-      if (!dataStorage.hasCollection(metadataCollectionName)) { // metadata collection does not exist
-         return false;
-      }
-
-      List<DataDocument> newAttributeInfo = dataStorage.run(queryCollectionAttributeInfo(collectionName, newName));
-
-      // check if the attribute with new name already exists in the collection
-      if (!newAttributeInfo.isEmpty()) {
+   public void renameAttribute(String collectionName, String oldName, String newName) throws AttributeAlreadyExistsException {
+      if (getAttributesNames(collectionName).contains(newName)) {
          throw new AttributeAlreadyExistsException(ErrorMessageBuilder.attributeAlreadyExistsString(newName, collectionName));
```

Comment:
+1 on moving this check at the beginning of the method and putting the rest of the code outside the `if` statement. It should always be done this way in order to avoid unnecessary nesting.